projects/vdk-plugins/vdk-impala/tests/functional/test_impala_lineage.py

```diff
@@ -0,0 +1,363 @@
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_funcs import jobs_path_from_caller_directory
+
+VDK_DB_DEFAULT_TYPE = "VDK_DB_DEFAULT_TYPE"
+VDK_IMPALA_HOST = "VDK_IMPALA_HOST"
+VDK_IMPALA_PORT = "VDK_IMPALA_PORT"
+VDK_IMPALA_SYNC_DDL = "VDK_IMPALA_SYNC_DDL"
+VDK_IMPALA_QUERY_POOL = "VDK_IMPALA_QUERY_POOL"
+
+
+class TestConfigPlugin:
+    def __init__(self, lineage_logger: ILineageLogger):
+        self.lineage_logger = lineage_logger
+
+    @hookimpl
+    def vdk_initialize(self, context):
+        # register a lineage logger which will be used to validate if correct lineage data was emitted
+        context.state.set(LINEAGE_LOGGER_KEY, self.lineage_logger)
+
+
+def execute_query(runner, query: str):
+    result: Result = runner.invoke(["impala-query", "--query", query])
+    cli_assert_equal(0, result)
+
+
+@pytest.mark.usefixtures("impala_service")
+class ImpalaLineageTest(TestCase):
+    @patch.dict(
+        os.environ,
+        {
+            VDK_DB_DEFAULT_TYPE: "IMPALA",
+            VDK_IMPALA_HOST: "localhost",
+            VDK_IMPALA_PORT: "21050",
+        },
+    )
+    def test_lineage_not_collected_for_invalid_query(self):
+        mock_lineage_logger = mock.MagicMock(ILineageLogger)
+        runner = CliEntryBasedTestRunner(
+            TestConfigPlugin(mock_lineage_logger), impala_plugin
+        )
+        result: Result = runner.invoke(
+            ["run", jobs_path_from_caller_directory("sql-job-syntax-error")]
+        )
+        cli_assert_equal(1, result)
+        assert not mock_lineage_logger.send.called
+
+    @patch.dict(
+        os.environ,
+        {
+            VDK_DB_DEFAULT_TYPE: "IMPALA",
+            VDK_IMPALA_HOST: "localhost",
+            VDK_IMPALA_PORT: "21050",
+        },
+    )
+    def test_lineage_for_create_and_insert_queries(self):
+        mock_lineage_logger = mock.MagicMock(ILineageLogger)
+        runner = CliEntryBasedTestRunner(
+            TestConfigPlugin(mock_lineage_logger), impala_plugin
+        )
+        result: Result = runner.invoke(
+            ["run", jobs_path_from_caller_directory("sql-job")]
+        )
+        cli_assert_equal(0, result)
+
+        # the query is different on every run of the test
+        class InsertLineageDataMatcher:
+            def __eq__(self, lineage_data: LineageData):
+                assert "INSERT INTO stocks VALUES" in lineage_data.query
+                assert lineage_data.query_type is None
+                assert lineage_data.query_status == "OK"
+                assert lineage_data.output_table.catalog is None
+                assert lineage_data.output_table.schema == "default"
+                assert lineage_data.output_table.table == "stocks"
+                assert len(lineage_data.input_tables) == 0  # no inputs for insert query
+                return True
+
+        # create table and select 1 queries should not result in sending lineage, hence test with called_once to verify
```

Comment:
How many `select 1` queries are to be expected in this test case?
Aren't the quantities of `select 1` correlated to the session length, the time window a data job does active DB work?

Reply:
I think the amount of select 1 queries is linearly correlated to the amount of sql queries executed.

Reply:
Is it documented like a common knowledge somewhere? Did you read and where from / test it ?

Reply:
Nope. It is my assumption based on the logs and that I observe them before queries are being executed (apart from the very first query executed on fresh DB connection).
I was hoping that the VDK team have better understanding of it.